Coding/fw/lib/WebServer/CaptivePortal.cpp

```diff
@@ -0,0 +1,172 @@
+    */
+    void handleRequest(AsyncWebServerRequest *request) override
+    {
+        if (nullptr == request)
+        {
+            return;
+        }
+        /** Forces authentication */
+        if (!request->authenticate(WebConfig::getWEB_USER().c_str(), WebConfig::getWEB_PASS().c_str()))
+        {
+            return request->requestAuthentication();
+        }
+
+        if (HTTP_POST == request->method())
+        {
+            if (request->args() != 0)
+            {
+                int params = request->params();
+                for (int i = 0; i < params; i++)
+                {
+                    AsyncWebParameter *p = request->getParam(i);
+                    credentialsProcessor(p->name(), p->value());
+                }
+                request->send(SPIFFS, "/setCredentials.html", String(), false, captivePageProcessor);
+                reqRestart();
+            }
+        }
+        else if (HTTP_GET == request->method())
+        {
+            request->send(SPIFFS, "/STACredentials.html", String(), false, captivePageProcessor);
+        }
+        else
+        {
+            request->send(400, "text/html", "\n Error. Bad Request");
+        }
+    }
+
+    bool isRequestHandlerTrivial() override
+    {
+        return false;
+    }
+
+private:
+    static String captivePageProcessor(const String &var)
+    {
+        String temp;
+        return temp;
+    }
+};
+
+/* PUBLIC METHODES ********************************************************************************/
+
+/* PROTECTED METHODES *****************************************************************************/
+
+/* PRIVATE METHODES *******************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+
+/* VARIABLES **************************************************************************************/
+static CaptiveRequestHandler CaptivePortalReqHandler; /**< Instance of Handler */
+static bool restartRequested = false; /**<  Variable to call Restart */
+
+/* EXTERNAL FUNCTIONS *****************************************************************************/
+/**
+ * @brief Links the Captive Portal handler to the Server
+ * 
+ * @param server AsyncWebserver Instance to initialize to
+ */
+void CaptivePortal::init(AsyncWebServer &server)
+{
+    server.serveStatic("/css/w3.css", SPIFFS, "/css/w3.css", "max-age = 3600");
+    server.serveStatic("/pictures/NewTec_Logo.png", SPIFFS, "/pictures/NewTec_Logo.png", "max-age = 3600");
+    server.addHandler(&CaptivePortalReqHandler).setFilter(ON_AP_FILTER);
+}
+
+/**
+ * @brief return true if restart has been requested 
+ */
+bool CaptivePortal::isRestartRequested()
+{
+    return restartRequested;
+}
+
+/* INTERNAL FUNCTIONS *****************************************************************************/
+
+/**
+ * Request restart.
+ */
+static void reqRestart()
```

Comment:
what is the added value of this static function  if all it does is changing a static bool flag ?